src/files/compose/validateDappnodeCompose.ts

```diff
@@ -0,0 +1,273 @@
+        aggregatedError.errors.push(
+          Error(
+            `The docker network ${networkName} is not allowed. Only docker networks ${params.DOCKER_WHITELIST_NETWORKS.join(
+              ","
+            )} are allowed`
+          )
+        );
+      // Check all networks are external
+      if (cpNetworks[networkName].external === false)
+        aggregatedError.errors.push(
+          Error(
+            `The docker network ${networkName} is not allowed. Docker internal networks are not allowed`
+          )
+        );
+    }
+  }
+}
+
+/**
+ * Ensures the compose keys are whitelisted
+ */
+function validateComposeServiceKeys(
+  compose: Compose,
+  cpServiceName: string
+): void {
+  const cpServiceKeys = Object.keys(compose.services[cpServiceName]);
+  for (const cpServiceKey of cpServiceKeys) {
+    if (!params.SAFE_KEYS.includes(cpServiceKey))
+      aggregatedError.errors.push(
+        Error(
+          `Compose service ${cpServiceName} has key ${cpServiceKey} that is not allowed. Allowed keys are: ${params.SAFE_KEYS.join(
+            ","
+          )}`
+        )
+      );
+  }
+}
+
+/**
+ * Ensures the compose keys values are valid for dappnode
+ */
+function validateComposeServiceValues(
+  compose: Compose,
+  isCore: boolean,
+  cpServiceName: string
+): void {
+  // Check that if defined, the DNS must be the one provided from the bind package
+  const cpServiceDns = compose.services[cpServiceName].dns;
+  if (cpServiceDns && !params.DNS_SERVICE.includes(cpServiceDns))
+    aggregatedError.errors.push(
+      Error(
+        `Compose service ${cpServiceName} has DNS different than ${params.DNS_SERVICE}`
+      )
+    );
+
+  // Check compose pid feature can only be used with the format service:*. The pid:host is dangerous
+  const cpServicePid = compose.services[cpServiceName].dns;
+  if (cpServicePid && !cpServicePid.startsWith("service:"))
+    aggregatedError.errors.push(
+      Error(
+        `Compose service ${cpServiceName} has PID feature differnet than service:*`
+      )
+    );
+
+  // Check only core packages cand be privileged
+  const cpServicePrivileged = compose.services[cpServiceName].privileged;
+  if (!isCore && cpServicePrivileged && cpServicePrivileged === true)
+    aggregatedError.errors.push(
+      Error(
+        `Compose service ${cpServiceName} has privileged as true but is not a core package`
+      )
+    );
+
+  // Check Only core packages can use network_mode: host
+  const cpServiceNetworkMode = compose.services[cpServiceName].network_mode;
+  if (!isCore && cpServiceNetworkMode && cpServiceNetworkMode === "host")
+    aggregatedError.errors.push(
+      Error(
+        `Compose service ${cpServiceName} has network_mode: host but is not a core package`
+      )
+    );
+}
+
+/**
+ * Ensure the compose services networks are whitelisted
+ */
+function validateComposeServiceNetworks(
+  compose: Compose,
+  isCore: boolean,
+  cpServiceName: string
```

Comment:
Why prefix with `cp`? Is this entire file there only one possible thing named `serviceName` remove the `cp` prefix from everywhere to make the code more readable